locksmith/src/controllers/purchaseController.ts

```diff
@@ -48,30 +42,48 @@ namespace PurchaseController {
     const { expiry } = req.body.message.purchaseRequest
     const { lock } = req.body.message.purchaseRequest
     const requestedPurchaseAmount = req.body.message.purchaseRequest.USDAmount
+    const purchaser = req.body.message.purchaseRequest.recipient
 
     if (expired(expiry)) {
       return res.sendStatus(412)
     }
 
-    const keyPricer = new KeyPricer(
-      config.web3ProviderHost,
-      config.unlockContractAddress
-    )
-
-    const currentPrice = await keyPricer.keyPriceUSD(lock)
-
+    const currentPrice = await keyPricer().keyPriceUSD(lock)
     const validRequestPrice = PriceRange.within({
       requestPrice: requestedPurchaseAmount,
       currentPrice,
     })
 
     if (validRequestPrice) {
-      return res.sendStatus(200)
+      const hash = await processor().initiatePurchaseForConnectedStripeAccount(
+        purchaser,
+        lock,
+        config.purchaserCredentails,
+        config.web3ProviderHost,
+        purchaser,
+        'connectedStripeAccount' // The expectation is this will come from an external data source
```

Comment:
Yes, it will come from the lock's public metadata! 
```suggestion
        'connectedStripeAccount' // TODO: replace with value coming from lock metadata, saved by one of the lock managers
```